### PR TITLE
Int 8940 make more efficient + Quality of life improvements

### PR DESCRIPTION
**Pull request by Gonzalo-Avalos-Ribas**
# Description

Thank you for contributing to a JupiterOne integration!

Please include a summary of the change and which issue is fixed. Please also
include relevant motivation and context. List any dependencies that are required
for this change.

## Type of change

Please leave any irrelevant options unchecked.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to
      not work as expected)
- [ ] This change requires a documentation update

### General Development Checklist:

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### Integration Development Checklist:

Please leave any irrelevant options unchecked.

- [ ] I have checked for additional permissions required to call any new API
      endpoints, and have documented any additional permissions in
      `jupiterone.md`, where necessary.
- [ ] My changes properly paginate the target service provider's API
- [ ] My changes properly handle rate limiting of the target service provider's
      API
- [ ] My new integration step is instrumented to execute in the correct order
      using `dependsOn`
- [ ] I have referred to the
      [JupiterOne data model](https://github.com/JupiterOne/data-model/tree/main/src/schemas)
      to ensure that any new entities/relationships, and relevant properties,
      match the recommended model for this class of data
- [ ] I have updated the `CHANGELOG.md` file to describe my changes
- [ ] When changes include modifications to existing graph data ingestion, I've
      reviewed all existing managed questions referencing the entities,
      relationships, and their property names, to ensure those questions still
      function with my changes.

### CODE REVIEW

File: src/client.ts [hunk 1]
```diff
@@ -20,7 +20,7 @@ export enum ServiceNowTable {
   SYS_DICTIONARY = 'sys_db_object',
 }
 
-const DEFAULT_RESPONSE_LIMIT = 100;
+const DEFAULT_RESPONSE_LIMIT = 500;
```
Comment (Gonzalo-Avalos-Ribas) [inline src/client.ts:23]: This should make all steps more efficient, without chaning too much the memory requirements.

File: src/client.ts [hunk 2]
```diff
@@ -119,10 +119,38 @@ export class ServiceNowClient {
...
+              const timeNow = new Date().getTime();
+              this.logger.info(
+                {
+                  TimeToWaitInMilis: retryAfter,
+                  timeNow: timeNow,
+                  waitingTill: timeNow + retryAfter,
+                },
+                'Hit Max API rate. Waiting for retry.',
+              );
+              await this.sleep(Number(timeToWaitInMillis));
```
Comment (Gonzalo-Avalos-Ribas) [inline src/client.ts:142]: In service now you can a custom API rate limit. This means this logic is needed in order to wait if we ever hit it.

File: src/steps/index.test.ts [hunk 3]
```diff
@@ -1,23 +1,11 @@
 import {
```
Comment (Gonzalo-Avalos-Ribas) [inline src/steps/index.test.ts:1]: Upgraded all old tests to the new ones.

File: src/client.ts (hunk 2 above)
Comment (RonaldEAM) [reply src/client.ts:142]: the `attempt` package has parameters set up for delay so we don't need to do it manually with sleep here, here's an example on how to do it https://github.com/JupiterOne/graph-tenable-io/blob/main/src/tenable/TenableClient.ts#L430

**Review by mishelashala — APPROVED** [review]

**Pull request merged**